content/learn/book/modular-architecture/project-organization.md

```diff
@@ -0,0 +1,107 @@
++++
+title = "Project Organization"
+insert_anchor_links = "right"
+[extra]
+weight = 1
++++
+
+As your Bevy project grows, storing it all in a single `main.rs` file will become frustrating!
+But how should you split it apart, and what are your tools for doing so?
+
+Thankfully, the answers to these questions are pretty simple:
+
+- split your code based on the "domain" it covers
+  - think about where you might want to reuse, or selectively disable bits of functionality
+  - code that changes together should live together
+  - splits like "world generation" and "combat" and "main menu" are great
+  - don't organize your modules by e.g. "components" and "systems"!
```

Comment:
I'm not terribly opinionated on this one, but for what its worth I have structured most of my Bevy projects this way & prefer to keep related components types together as I find that most components don't fit perfectly in most groupings.
On the other hand with systems, while I don't put them in a systems module, they do usually reside as private fns within a plugin module, except for the unusual system that is used across multiple plugins.

I suppose this structure may be controversial but I wanted to share my experience nonetheless for at least another data point to consider!

Reply:
I think that's a reasonable point; let me try to re-explain this. I've seen people with web backgrounds (I think) repeatedly trying to do `root/components` which is a complete disaster. 

I want to discourage that specifically!

Reply:
Ah that makes a lot of sense!